File: app/src/main/java/uit/dev/demomvp/Presenter/LoginLogicPresenter.java
package uit.dev.demomvp.Presenter;

import java.util.ArrayList;

import uit.dev.demomvp.Model.User;
import uit.dev.demomvp.View.LoginView;

/**
 * Created by Phi on 09/12/2017.
 */

public class LoginLogicPresenter implements LoginPresenter {
    LoginView loginView ;

    public  LoginLogicPresenter(LoginView loginView)
    {
        this.loginView = loginView;
    }

    @Override
    public void CheckLogin(String username, String password) {

        if ((username.equals("gkz@gmail.com")|| username.equals("gkz") )&& password.equals("123456"))
        {
            loginView.loginSuccess();
        }
        else
        {
            loginView.loginFail();
        }
    }

    @Override
    public void AddEmailToListview(String email, ArrayList<User> listUser) {
     User user = new User();
     user.setEmail(email);
     listUser.add(user);
    }
}

File: app/src/main/java/uit/dev/demomvp/Presenter/LoginLogicPresenter.kt
package uit.dev.demomvp.Presenter

import java.util.ArrayList

import uit.dev.demomvp.Model.User
import uit.dev.demomvp.View.LoginView

/**
 * Created by Phi on 09/12/2017.
 */

class LoginLogicPresenter(internal var loginView: LoginView) : LoginPresenter {

    override fun CheckLogin(username: String, password: String) {

        if ((username == "gkz@gmail.com" || username == "gkz") && password == "123456") {
            loginView.loginSuccess()
        } else {
            loginView.loginFail()
        }
    }

    override fun AddEmailToListview(email: String, listUser: ArrayList<User>) {
        val user = User()
        user.email = email
        listUser.add(user)
    }
}
